GBK Encoding on Windows 11 Causes `validate-pyproject` Error with UTF-8 File Containing Em Dash

I’m using Windows 11 with the system character encoding set to `GBK`, and my `pyproject.toml` file is encoded in `UTF-8`.

When a comment contains an em dash (`—`), `validate-pyproject` throws an error:
```
[ERROR] UnicodeDecodeError: 'gbk' codec can't decode byte 0x94 in position 41: illegal multibyte sequence
```

My `pyproject.toml` file:

```toml
[project]
name = "a"
version="1.0" # —
```

Entering Chinese characters will also cause the same error.

--- a/src/validate_pyproject/cli.py
+++ b/src/validate_pyproject/cli.py
@@ -52,7 +52,7 @@ def critical_logging() -> Generator[None, None, None]:
         raise
 
 
-_STDIN = argparse.FileType("r")("-")
+_STDIN = argparse.FileType("r", encoding="utf-8")("-")
 
 META: dict[str, dict] = {
     "version": dict(
@@ -64,7 +64,7 @@ META: dict[str, dict] = {
         dest="input_file",
         nargs="*",
         # default=[_STDIN],  # postponed to facilitate testing
-        type=argparse.FileType("r"),
+        type=argparse.FileType("r", encoding="utf-8"),
         help="TOML file to be verified (`stdin` by default)",
     ),
     "enable": dict(

